### ISSUE TITLE
jinja[spacing] warning with nested if statements

### ISSUE THREAD

**Issue by slovdahl** [ansible/ansible-lint#5146]
##### Summary

`ansible-lint` versions 24.12.2 and 26.6.0 incorrectly warn that spacing could be improved for this

```yaml
  - name: Check if the existing service is masked
    ansible.builtin.set_fact:
      existing_service_is_masked: >-
        {%- if existing_service_name in services_stop_existing.ansible_facts.services -%}
          {%- if services_stop_existing.ansible_facts.services[existing_service_name]['status'] == 'masked' -%}
            true
          {%- else -%}
            false
          {%- endif -%}
        {%- else -%}
          false
        {%- endif -%}
```

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
$ ansible-lint --version
ansible-lint 26.6.0 using ansible-core:2.16.19 ansible-compat:26.6.0 ruamel-yaml:0.18.16 ruamel-yaml-clib:0.2.15
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

```console
$ ansible-lint playbook.yml
```

`playbook.yml`:

```yaml
- name: Test playbook
  hosts: all
  tasks:
    - name: Check if the existing service is masked
      ansible.builtin.set_fact:
        existing_is_masked: >-
          {%- if existing_service_name in services_stop_existing.ansible_facts.services -%}
            {%- if services_stop_existing.ansible_facts.services[existing_service_name]['status'] == 'masked' -%}
              true
            {%- else -%}
              false
            {%- endif -%}
          {%- else -%}
            false
          {%- endif -%}
```

##### Desired Behavior

No warning and no change applied with `--fix`.

Possible security bugs should be reported via email to `security@ansible.com`

##### Actual Behavior

```console
$ ansible-lint playbook.yml
WARNING  Listing 1 violation(s) that are fatal
[jinja[spacing]](https://ansible.readthedocs.io/projects/lint/rules/jinja/): Jinja2 spacing could be improved: {%- if existing_service_name in services_stop_existing.ansible_facts.services -%}
  {%- if services_stop_existing.ansible_facts.services[existing_service_name]['status'] == 'masked' -%}
    true
  {%- else -%}
    false
  {%- endif -%}
{%- else -%}
  false
{%- endif -%} -> {%- if existing_service_name in services_stop_existing.ansible_facts.services -%}
  {%- if services_stop_existing.ansible_facts.services[existing_service_name]['status'] == 'masked' -%}
    true
{%- else -%}
    false
{%- endif -%}
{%- else -%}
  false
{%- endif -%}[/] (warning)
playbooks_local/test copy 2.yml:6 Jinja2 template rewrite recommendation: `{%- if existing_service_name in services_stop_existing.ansible_facts.services -%}
  {%- if services_stop_existing.ansible_facts.services[existing_service_name]['status'] == 'masked' -%}
    true
{%- else -%}
    false
{%- endif -%}
{%- else -%}
  false
{%- endif -%}`.[/]

Read [documentation](https://ansible.readthedocs.io/projects/lint/configuring/#ignoring-rules-for-entire-files) for instructions on how to ignore specific rule violations.

# Rule Violation Summary

  1 [jinja](https://ansible.readthedocs.io/projects/lint/rules/) profile:basic tags:formatting

Passed: 0 failure(s), 1 warning(s) on 1 files. Last profile that met the validation criteria was 'min'.
```

With `--fix` it is changed to something less readable:

```yaml
- name: Check if the existing service is masked
  ansible.builtin.set_fact:
    existing_service_is_masked: >-
      {%- if existing_service_name in services_stop_existing.ansible_facts.services -%}
        {%- if services_stop_existing.ansible_facts.services[existing_service_name]['status'] == 'masked' -%}
          true
      {%- else -%}
          false
      {%- endif -%}

      {%- else -%}
        false
      {%- endif -%}
```

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve
